Sources/Tracking/CustomerIO.swift

```diff
@@ -35,15 +35,25 @@ public extension CustomerIO {
         configure configureHandler: ((inout SdkConfig) -> Void)?
     ) {
         var sdkConfig = SdkConfig.Factory.create(siteId: siteId, apiKey: apiKey, region: region)
+        let newDiGraph = DIGraph(sdkConfig: sdkConfig)
+        let implementation = DataPipeline.initialize(moduleConfig: DataPipelineConfigOptions.Factory.create(sdkConfig: sdkConfig))
+        let profileStore = newDiGraph.profileStore
+
+        // This code handles the scenario where a user migrates
+        // from the Journeys module to the CDP module while already logged in.
+        // This ensures the CDP module is informed about the
+        // currently logged-in user for seamless processing of events.
+        if DataPipeline.shared.analytics.userId == nil {
+            if let identifier = profileStore.identifier {
+                DataPipeline.shared.identify(identifier: identifier, body: [:])
```

Comment:
We should clear `profileStore.identifier` after setting it for `analytics` to make sure the code only run once.

Use case in my that could break things:

- Identify user using old SDK
- Update SDK to use `DataPipeline` -> The profile should be identified in `DataPipeline`
- Clear identify using new SDK
- Relaunch the app so SDK initialization is called again -> The will re-identify with old id in `DataPipeline` even though it was cleared before relaunch

Reply:
Great catch on the use case, Rehan ! I have made the necessary updates and you can find the changes i this [commit](https://github.com/customerio/customerio-ios/pull/490/commits/4aee0591e4b9c4624a4f100484de0d7a9ca0b62e).

Reply:
I think a test should be written to confirm this use case is handled.